its/src/test/java/com/sonar/it/csharp/TestProjectTest.java

```diff
@@ -53,10 +54,19 @@ public void projectIsAnalyzed() {
   @Test
   public void logsContainInfoAndWarning() {
     assertThat(buildResult.getLogs()).contains(
-      "This sensor will be skipped, because the current solution contains only TEST files and no MAIN files. " +
+      "This C# sensor will be skipped, because the current solution contains only TEST files and no MAIN files. " +
         "Your SonarQube/SonarCloud project will not have results for C# files. " +
         "Read more about how the SonarScanner for .NET detects test projects: https://github.com/SonarSource/sonar-scanner-msbuild/wiki/Analysis-of-product-projects-vs.-test-projects",
-      "Found 1 MSBuild project. 1 TEST project."
+      "Found 1 MSBuild C# project: 1 TEST project."
     );
+    verifyGuiAnalysisWarning(buildResult);
+  }
+
+  // Verifies the analysis warning is raised inside SQ
+  private void verifyGuiAnalysisWarning(BuildResult buildResult) {
```

Comment:
This method appears 3 times in the project with small variations (at the first look only the language name is different). I suggest to extract and reuse it.